cluster/common.sh

```diff
@@ -96,11 +96,15 @@ function create-kubeconfig() {
       cluster_args=(
           "--server=${KUBE_SERVER:-https://${PROXY_RESERVED_IP}}:443"
        )
-    elif [[ "${KUBERNETES_TENANT_PARTITION:-false}" == "true" ]] || [[ "${KUBERNETES_RESOURCE_PARTITION:-false}" == "true" ]]; then
+    elif [[ "${ENABLE_APISERVER_INSECURE_PORT:-false}" == "true" ]]; then
```

Comment:
We need a new var here instead of overloading ENABLE_APISERVER_INSECURE_PORT. Perhaps ENABLE_INSECURE_CLUSTER_MODE, which sets ENABLE_APISERVER_INSECURE_PORT to true in config-default.sh. This is because we can have ENABLE_APISERVER_INSECURE_PORT=true for debug/metric purposes  while still having secure communication between components.

Reply:
good point. originally i thought this can be done later but it seem Prometheus etc currently ( in our code ) does not connect to the secure api servers. i will take a look on to to address it.

Reply:
this was fixed, now we have removed ENABLE_APISERVER_INSECURE_PORT and with two flags in the system:
1. KUBE_ ENABLE_APISERVER_INSECURE_PORT only controls if the 8080 insecure port is opened or not when starting api server
2. USE_APISERVER_INSECURE_PORT controls the kubeconfig generation either as a insecure one or secure one ( with https, 443 port, token and certs in it )